### ISSUE TITLE
Brightness OSD syncing issues

### ISSUE THREAD

**Issue by WatchMkr** [pop-os/cosmic-osd#24]
To reproduce:

- Use the battery applet to increase LCD brightness to 100%
- Attempt to use the brightness up hotkey
  - The OSD does not appear
- Use the brightness down hotkey 
  - The OSD will appear and the value will be low (17-20%)
 - Brightness up hotkey will work again until at 100% (but it's only around 19% on the OSD)

Tested on the darp8 with open firmware.

**Comment by XV-02** [comment]
Having looked at other systems, this appears to be a general issue. 

On Pang12, 100% screen brightness reports as "252" , with the bar showing as "full" for any value of 100 or greater. On Pang12, changing brightness was more responsive than other systems I tested. When I adjust brightness, the OSD shows an immediate change and I can rapidly increase and decrease brightness. The numeric value reported corresponds to the `actual_brightness` and `brightness` values for the display in sysfs (`/sys/class/backlight/amdgpu_bl1` for Pang12)


On Intel system I've tested - Addw3, Oryp7(AMOLED), and Darp8 - 100% screen brightness (when `cat`ing values in `/sys/class/backlight/acpi_video0`) reports as "19". Much like the Pang12, the value reported by the OSD corresponded to this value - so the OSD reports 19% brightness.

On recent Intel systems (I haven't tested older systems) each step is ennumerated, rather than corresponding to what looks to be a 8-bit unsigned value for the AMD equivalent. 

On those Intel systems, there is a 0.5-1.0 second delay between any brightness change and the OSD updating. Not between the change being made and that change being displayed, but between an input and the OSD updating, as two quick step changes will result in the OSD reporting a jump in value of 2, not two jumps in value of 1. 

Additionally, changing brightness on Intel systems seem to be very slow, with the brightness setting only moving a single increment every .5 seconds or so. This seems to be somewhat tied to either OSD or Cosmic broadly. Gnome on the same systems hasn't got the same lag, and if I directly change the value by clicking the brightness slider in the power applet (but not dragging the slider) I get instant changes. 

As an side note, we can't merely report brightness as `actual_brightness`/`max_brightness` values from the sysfs, as the Pang12 never gets above 252 for `actual_brightness` but reports 255 for `max-brightness` in my testing. This means that, even with rounding, we'd never be above 99% brightness.
